### ISSUE TITLE
Condor job classads required for monitoring

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#8958]
As a follow up of:
https://its.cern.ch/jira/browse/CMSMONIT-41
https://its.cern.ch/jira/browse/CMSMONIT-61

we need to implement/set the following classads:
* `CMS_WMTool`: with `WMAgent` value
* `CMS_SubmissionTool`: again with `WMAgent`
* `CMS_Type`: likely the same as the dashboard activity used to be. One of the following values are allowed: `(Production, Test, Tier0)`

others that will require a bit more work:
* `CMS_TaskType`: more or less the same logic as used by Benjamin's script. See https://github.com/dmwm/WMCore/issues/8802 for further details
* `CMS_PrimaryInputLocation`: "did this job ask ro local or WAN reads ? (what it got may be different, but we can maybe tell)". Values are: `(Onsite, Offsite, Mixed)`
* `CMS_SecondaryInputLocation`: "did this job ask ro local or WAN reads ? (what it got may be different, but we can maybe tell)". Values are: `(Onsite, Offsite, Mixed)`

Schema is at: https://github.com/dmwm/CMSMonitoring/blob/master/schemas/ClassAds.json

The following jobads are already in place: `CMS_WMTool`, `CMS_SubmissionTool`, `CMS_Type`

**Comment by amaltaro** [comment]
Jean-Roch, confusing in which sense?

I also have a question, are there any campaigns that only read data remotely (either primary or secondary). Or those workflows with TrustPUSitelists/TrustSitelists always have a few sites whitelisted that also contain part of the data? Just trying to see where we have more "offsite" compared to "mixed" or if it's the inverse. It's going to be tricky to get it properly done in the agent...

**Comment by vlimant** [comment]
confusing in the sense that you might end up with a job reading data fully locally and still having CMS_PrimaryInputLocation = Offsite

**Comment by vlimant** [comment]
for campaign settings, in some sub-set of setting, it might be all in one case, but only in some small sub-set

**Comment by vlimant** [comment]
and yes ; to do it right in the agent at job submission time is going to be tricky, as the what can actually happen may be different..
To get it "less wrong", you might need to make it an expression that takes into account the known data location at submission time, like CMS_PrimaryInputLocation = "GlideinSite in CMS_DataLocationSites" (whichever is the actual syntax)

**Comment by vkuznet** [comment]
Alan, please add those attributes to FWJR docs send to WMArchive too since we need them there as well. But I suggest to not use CMS_ prefix since we don't use it in WMArchvie, so instead of `CMS_PrimaryInputLocation` please add `primaryInputLocation`, etc.

For your convenience, here is current WMArchive schema:
https://github.com/dmwm/WMArchive/blob/master/src/python/WMArchive/Schemas/FWJRProduction.py

If you agree to send them then I can go ahead and change WMArchive schema accordingly. Since we used nested structure we need to properly find their places in WMArchive schema.

**Comment by amaltaro** [comment]
We have delivered some of the features/classads requested in this issue and I feel that we should actually create a/some specific GH issue (and/or further discuss it) for what is still desired to have as a job classad.

In addition, this issue aggregates much of this monitoring-related information:
https://github.com/dmwm/WMCore/issues/9455

and we might want to create new issues and turn them sub-tasks of #9455. Closing this one out.
